swarm/storage/localstore.go

```diff
@@ -27,9 +29,10 @@ type LocalStore struct {
 	DbStore  ChunkStore
 }
 
-// This constructor uses MemStore and DbStore as components
-func NewLocalStore(hash SwarmHasher, params *StoreParams) (*LocalStore, error) {
-	dbStore, err := NewDbStore(params.ChunkDbPath, hash, params.DbCapacity, params.Radius)
+// This constructor uses MemStore and DbStore as components.
+// If mockStore is not nil, it will be used by DbStore to store chunk data.
+func NewLocalStore(hash SwarmHasher, params *StoreParams, mockStore *mock.NodeStore) (*LocalStore, error) {
+	dbStore, err := NewMockDbStore(params.ChunkDbPath, hash, params.DbCapacity, params.Radius, mockStore)
```

Comment:
I understand this to mean we use `Mock` inside naming for production code. I understand this is because of the implementation needs nevertheless I wonder if we could use something like a factory pattern, an interface or something to avoid this?

Reply:
Abstraction is already tried with LocalStore, DbStore, DbAcceess and it requires significant changes to storage and network packages. Injecting mock store here is more clear and requires no rewriting of existing code.